nomad/structs/devices.go

```diff
@@ -0,0 +1,130 @@
+package structs
+
+// DeviceAccounter is used to account for device usage on a node. It can detect
+// when a node is oversubscribed and can be used for deciding what devices are
+// free
+type DeviceAccounter struct {
+	Devices map[DeviceIdTuple]*DeviceAccounterInstance
+}
+
+// DeviceAccounterInstance wraps a device and adds tracking to the instances of
+// the device to determine if they are free or not.
+type DeviceAccounterInstance struct {
+	// Device is the device being wrapped
+	Device *NodeDeviceResource
+
+	// Instances is a mapping of the device IDs of the instances to their usage.
+	// Only a value of 0 indicates that the instance is unused.
+	Instances map[string]int
```

Comment:
what's the difference between the ids in this map and the device IDs in the wrapper `DeviceAccounter`? Having a clarifying comment would be helpful.